src/engraving/rendering/dev/tremololayout.cpp

```diff
@@ -344,10 +344,11 @@ void TremoloLayout::createBeamSegments(Tremolo* item, LayoutContext& ctx)
     double bboxTop = item->up()
                      ? std::min(mainStroke->line.y1(), mainStroke->line.y2())
                      : std::max(mainStroke->line.y1(), mainStroke->line.y2());
-    double halfWidth = ctx.conf().styleMM(Sid::beamWidth).val() / 2. * (item->up() ? -1. : 1.);
+    double width = ctx.conf().styleMM(Sid::beamWidth).val();
+    double halfWidth = width / 2. * (item->up() ? -1. : 1.);
 
     RectF bbox = RectF(mainStroke->line.x1(), bboxTop + halfWidth, mainStroke->line.x2() - mainStroke->line.x1(),
-                       std::abs(mainStroke->line.y2() - mainStroke->line.y1()) - halfWidth * 2.);
+                       std::abs(mainStroke->line.y2() - mainStroke->line.y1()) + width);
```

Comment:
It's the right idea, but it's not 100% correct, see:
<img width="286" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/824d1cad-23d9-4584-8caa-860c55ec4611">

It can be done more easily: there is really no need for the bbox calculation to be aware of up/down. Try this, which should work and looks clearer :)

```
LineF line = mainStroke->line;
double halfWidth = 0.5 * ctx.conf().styleMM(Sid::beamWidth).val();
double bboxTop = std::min(line.y1(), line.y2()) - halfWidth;
double bboxBottom = std::max(line.y1(), line.y2()) + halfWidth;

RectF bbox = RectF(line.x1(), bboxTop, line.x2() - line.x1(), bboxBottom - bboxTop);
```

Reply:
Oh that makes so much sense. I had no idea why bboxTop was being calculated like that. I assumed it was correct